Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -349,6 +350,30 @@ void ValidateProducedLayout()
             }
         }
 
+        [Feature(Features.Symlink)]
+        [Feature(Features.OpaqueDirectory)]
+        [TheoryIfSupported(requiresSymlinkPermission: true)]
+        public void DirectorySymlinksInOutputDirectoryShouldSuccess()
+        {    
+            AbsolutePath rootDirAbsPath = CreateUniqueObjPath($"{SealDirectoryKind.SharedOpaque}.framework");
+            string rootDir = rootDirAbsPath.ToString(Context.PathTable);
+            System.Diagnostics.Debugger.Launch();
+
+            // schedule producer pip: produce opaque directory with a bunch of symlinks
+            var operations = GetLayoutProducingOperationsWithDummyReadFile(rootDir, "producer");
+            var producerPipBuilder = CreatePipBuilder(operations);
+            producerPipBuilder.ToolDescription = StringId.Create(Context.StringTable, "producer");
+
+            // if dirKind is an opaque output, then just declare the root directory as an opaque output;
+            // otherwise, add all outputs explicitly and schedule a seal directory pip to seal them.
+            Process producerPip;
+            DirectoryArtifact outputDirArtifact;
+            producerPipBuilder.AddOutputDirectory(rootDirAbsPath, kind: SealDirectoryKind.SharedOpaque);
+            producerPip = SchedulePipBuilder(producerPipBuilder).Process;
+            outputDirArtifact = producerPip.DirectoryOutputs.First();
+            ScheduleRunResult firstRunResult = RunScheduler().AssertSuccess();         
+        }
+
         [Feature(Features.Symlink)]

```

Comment:
I think the bug is not about making directory symlink work on Windows. The fix for the bug is simply to avoid catastrophic failure. Otherwise this will not be a bug, but rather a feature request.

As I mentioned in the other comment, we have two choices:
- Fail the pip and log error if the pip produces a directory symlink.
- Warn, but make the pip uncacheable. 
I prefer the latter.